packages/react-dom/src/client/ReactDOMHostConfig.js

```diff
@@ -971,8 +971,8 @@ export function didNotMatchHydratedContainerTextInstance(
   textInstance: TextInstance,
   text: string,
   isConcurrentMode: boolean,
+  shouldWarnDev: boolean,
 ) {
-  const shouldWarnDev = true;
```

Comment:
I wonder why it was written this way in the first place. I thought in the past that it's intentional but I don't remember why. Maybe it wasn't.

Reply:
@acdlite mind checking this out and seeing if there was a reason you had dev warnings hardcoded to `true` in https://github.com/facebook/react/pull/23364 ?

From what I can tell there was nothing specific to the test case that would have made this preferential

Reply:
Not that I can recall, the fix seems right to me

Reply:
👍